addons/mail/static/src/components/discuss/tests/discuss_tests.js

```diff
@@ -4262,6 +4268,31 @@ QUnit.test('mark channel as seen if last message is visible when switching chann
     );
 });
 
+QUnit.test('add custom filter should filter messages accordingly to selected filter', async function (assert) {
+    assert.expect(4);
+
+    this.data['mail.channel'].records.push({
+        id: 20,
+        name: "General"
+    });
+    await this.start({
+        async mockRPC(route, args) {
+            if (args.method === 'message_fetch') {
+                const domainsAsStr = args.kwargs.domain.map(domain => domain.join(''));
+                assert.step(`message_fetch_${domainsAsStr.join('')}`);
+            }
+            return this._super(...arguments);
+        },
+    });
+    assert.verifySteps(['message_fetch_needaction=true'], "A message_fetch request should have been done for needaction messages");
+
+    // Open filter menu of control panel and select a custom filter (id = 0)
+    await toggleFilterMenu(document.body);
+    await toggleAddCustomFilter(document.body);
+    await applyFilter(document.body);
+    assert.verifySteps(['message_fetch_id=0needaction=true'], "A message_fetch request should have been done for selected filter");
```

Comment:
also where does `needaction=true` come from? worth a comment that it is coming from inbox being selected by default, I assume?

Reply:
You resolved this and pushed changes but I didn't see answer to my message or any explanation added in code?

Reply:
I changed the assert comment : 

```js
assert.verifySteps(['message_fetch_needaction=true'], "A message_fetch request should have been done for needaction messages as inbox is selected by default");
```

Reply:
Ok I see it's posted as initial assert actually, I would still clarify comment here that it should be selected filter + domain of current thread (or however we can call that)